cmd/thanos/query_frontend.go

```diff
@@ -217,6 +223,13 @@ func runQueryFrontend(
 	cfg *queryFrontendConfig,
 	comp component.Component,
 ) error {
+	cfg.ForwardHeaders = append(cfg.ForwardHeaders, tenancy.DefaultTenantHeader)
+	cfg.orgIdHeaders = append(cfg.orgIdHeaders, tenancy.DefaultTenantHeader)
+	if cfg.TenantHeader != "" && cfg.TenantHeader != tenancy.DefaultTenantHeader {
```

Comment:
should this be an `if else`? or is it a case that we always want to forward the default tenant header since it is appended above?

Reply:
I prefer to use only `if` with a good initialization instead of `if else`. Makes it easier to read: "we always forward the default tenant header and **if** there's a custom one, we forward it too".

Buuuut, I think I can remove `cfg.ForwardHeaders = append(cfg.ForwardHeaders, cfg.TenantHeader)`, as a custom tenant header doesn't need to be forwarded as-is downstream. It'll be converted to the internal one (the default), before being forwarded.

But I still have to add it to the list of `orgIdHeaders`.

Reply:
yes, don't disagree, was just wondering if it was a case of always passing on the default but that answers it.